bug: a typed parameter `Identifier` types `typeAnnotation` as `null`

### Summary

A parameter `Identifier` (`ESTree.BindingIdentifier`) types its `typeAnnotation` as `null`, so a typed parameter's type annotation can't be read type-safely — accessing it requires an `as` cast.

For a constructor/function parameter such as `id: string`:

- **Expected**: `param.typeAnnotation` is typed `TSTypeAnnotation | null` (a typed parameter does have a `TSTypeAnnotation` at runtime). `FormalParameterRest.typeAnnotation` is already typed this way.
- **Current**: `BindingIdentifier.typeAnnotation` is typed `null` (`typeAnnotation?: null`), so it narrows to `null | undefined` and the annotation is unreadable without a cast.

Would it be possible to type `BindingIdentifier.typeAnnotation` as `TSTypeAnnotation | null`, matching `FormalParameterRest`?

### Reproduction


```ts
import type { ESTree } from "@oxlint/plugins";

// e.g. the `id` of `constructor(scope: Construct, id: string)`
function readIdType(param: ESTree.BindingIdentifier) {
  // `typeAnnotation?: null` in the type, so this is `null | undefined`
  const ann = param.typeAnnotation;
  return ann?.typeAnnotation.type; // TS2339: Property 'typeAnnotation' does not exist on type 'null'
}
```

Workaround in our rules:

```ts
const typeAnnotation = param.typeAnnotation as ESTree.TSTypeAnnotation | null;
```


### Environment

- corsa-oxlint: 0.12.0
- @oxlint/plugins: 1.66.0
- @corsa-bind/napi: 0.12.0
- @typescript/native-preview (tsgo): 7.0.0-dev.20260516.1
- Node.js: 24.14.1
- OS: macOS (darwin 23.1.0)

### Logs or output

```shell
error TS2339: Property 'typeAnnotation' does not exist on type 'null'.
```

--- a/src/bindings/nodejs/corsa_oxlint/ts/api_surface.test.ts
+++ b/src/bindings/nodejs/corsa_oxlint/ts/api_surface.test.ts
@@ -1,4 +1,5 @@
-import { describe, expect, it } from "vitest";
+import { describe, expect, expectTypeOf, it } from "vitest";
+import type { ESTree as OxlintESTree } from "@oxlint/plugins";
 
 import * as astUtilsEntry from "./ast_utils";
 import * as main from "./index";
@@ -24,6 +25,12 @@ describe("api surface", () => {
     );
     expect(main.TSESTree.AST_TOKEN_TYPES.Block).toBe("Block");
     expect(tsestreeEntry.AST_NODE_TYPES.Identifier).toBe("Identifier");
+  });
+
+  it("wires typed parameter identifiers to type annotations", () => {
+    expectTypeOf<OxlintESTree.BindingIdentifier["typeAnnotation"]>().toEqualTypeOf<
+      OxlintESTree.TSTypeAnnotation | null | undefined
+    >();
   });
 
   it("re-exports typescript-eslint-style utility namespaces", () => {

--- a/src/bindings/nodejs/corsa_oxlint/ts/index.ts
+++ b/src/bindings/nodejs/corsa_oxlint/ts/index.ts
@@ -1,3 +1,5 @@
+import "./oxlint_plugins";
+
 export { AST_NODE_TYPES, AST_TOKEN_TYPES, TSESTree } from "./compat";
 export * as ASTUtils from "./ast_utils";
 export * as JSONSchema from "./json_schema";

--- a/src/bindings/nodejs/corsa_oxlint/ts/oxlint_plugins.ts
+++ b/src/bindings/nodejs/corsa_oxlint/ts/oxlint_plugins.ts
@@ -1,0 +1,9 @@
+declare module "@oxlint/plugins" {
+  namespace ESTree {
+    interface BindingIdentifier {
+      typeAnnotation?: TSTypeAnnotation | null;
+    }
+  }
+}
+
+export {};

